File: Sample.DigitalNotice/Sample.DigitalNotice.Web/ClientApp/src/common/entities/Map.ts
import { NoticeStatus } from '../enums/NoticeStatus'
import { NoticeType } from '../enums/NoticeType'
import { MapItem } from './MapItem'
import { TemplateItem } from './TemplateItem'

export type Map = {
  id?: string
  name: string
  description: string
  createdDate?: string
  status: NoticeStatus
  type: NoticeType
  templateItems: TemplateItem[]
  withNameColumn: boolean
  items: MapItem[]
}

File: Sample.DigitalNotice/Sample.DigitalNotice.Common/Entities/Map.cs
namespace Sample.DigitalNotice.Common.Entities;

/// <summary>
/// Represents a map derived from a notice.
/// </summary>
public class Map : Notice
{
    /// <summary>
    /// Gets or sets the items associated with the map.
    /// </summary>
    public IEnumerable<MapItem> Items { get; set; }

    /// <summary>
    /// Gets or sets the template items associated with the map.
    /// </summary>
    public IEnumerable<TemplateItem> TemplateItems { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether the map has a name column.
    /// </summary>
    public bool WithNameColumn { get; set; }
}
